finesse/gui/interferometer_monitor.py

```diff
@@ -48,55 +67,59 @@ def __init__(self) -> None:
+        """Creates layouts to house the widgets."""
+        self._poll_wid_layout = QHBoxLayout()
+        self._prop_wid_layout = QGridLayout()
+
+        top = QWidget()
+        top.setLayout(self._prop_wid_layout)
+        bottom = QWidget()
+        bottom.setLayout(self._poll_wid_layout)
+
+        self._layout = QVBoxLayout()
+        self._layout.addWidget(top)
+        self._layout.addWidget(bottom)
+
+    def _create_prop_widgets(self, prop) -> Tuple[QLabel, QLineEdit]:
+        """Creates the widgets for displaying the monitored properties."""
+        prop_label = QLabel(prop.name)
+        val_lineedit = QLineEdit()
+        val_lineedit.setText(prop.val_str())
+        val_lineedit.setReadOnly(True)
+        val_lineedit.setAlignment(Qt.AlignmentFlag.AlignCenter)
+        return prop_label, val_lineedit
+
     def _display_props(self) -> None:
         """Creates and populates the widgets to view the EM27 properties."""
         for prop in self._data_table:
-            if prop.name not in self._prop_names:
-                # Update list of monitored properties and create corresponding label
-                self._prop_names.append(prop.name)
-                prop_label = QLabel(prop.name)
-                self._prop_labels.append(prop_label)
-
-                # Remove poll server label and icon before adding new property widgets
-                poll_server_label = self._layout.itemAtPosition(
-                    self._num_props, 0
-                ).widget()
-                self._layout.removeWidget(poll_server_label)
-                self._layout.removeWidget(self._poll_light)
-
-                self._layout.addWidget(prop_label, self._num_props, 0)
-
-                val_lineedit = QLineEdit()
-                val_lineedit.setText(f"{prop.value:.6f} {prop.unit}")
-                val_lineedit.setReadOnly(True)
-                val_lineedit.setAlignment(Qt.AlignmentFlag.AlignCenter)
-                self._val_lineedits.append(val_lineedit)
-                self._layout.addWidget(val_lineedit, self._num_props, 1)
-
-                # Add poll server label and icon to bottom
-                self._layout.addWidget(poll_server_label, self._num_props + 1, 0)
-                self._layout.addWidget(self._poll_light, self._num_props + 1, 1)
-
-                self._num_props += 1
+            if prop.name not in self._val_lineedits:
+                num_props = len(self._val_lineedits)
+
+                prop_label, val_lineedit = self._create_prop_widgets(prop)
+                self._val_lineedits[prop.name] = val_lineedit
+
+                self._prop_wid_layout.addWidget(prop_label, num_props, 0)
+                self._prop_wid_layout.addWidget(val_lineedit, num_props, 1)
             else:
-                idx = self._prop_names.index(prop.name)
-                self._val_lineedits[idx].setText(f"{prop.value:.6f} {prop.unit}")
+                self._val_lineedits[prop.name].setText(prop.val_str())
```

Comment:
It's good that you've encapsulated the creation of the prop widgets, but I think the adding of the widgets to the layout really belongs in there too.

And I think this can be encapsulated a bit further so that this block doesn't have to worry about finding and creating the relevant widgets at all, just setting the label. (I had to scrutinise the code to check that you were setting the value of the `QLineEdit` correctly on both branches of the `if`-statement as it is.)

What I was suggesting is that you could have a function called `_get_prop_lineedit()` or something and then this block would look like:

```py
for prop in self._data_table:
    lineedit = _get_prop_lineedit(prop.name)
    lineedit.setText(prop.val_str())
```

If the relevant `QLineEdit` already exists, then `_get_prop_lineedit()` will just return it, otherwise it will create it and then return it.